{{project-name}}/build.rs

```
@@ -93,9 +71,12 @@ fn main() {
 
         cmd.env("CARGO_CFG_BPF_TARGET_ARCH", arch);
 
-        // Workaround to make sure that the rust-toolchain.toml is respected.
-        for key in ["RUSTUP_TOOLCHAIN", "RUSTC"] {
-            cmd.env_remove(key);
+        // Workaround to allow cargo-in-cargo to work.
+        for (key, value) in env::vars_os() {
+            use std::os::unix::ffi::OsStrExt as _;
+            if key.as_bytes().starts_with(b"RUST") || key.as_bytes().starts_with(b"CARGO") {
```

isn't this confusing if one intentionally sets RUSTFLAGS?

---

Are you setting RUSTFLAGS for the cargo-in-cargo? What for?

---

I'm on @tamird's side here for one another reason - if someone sets `RUSTFLAGS=-C linker=clang -C link-arg=-fuse-ld=[lld/mold]`, we don't want to pass that to the eBPF crate.

---

>  Are you setting RUSTFLAGS for the cargo-in-cargo? What for?

I'm doing `RUSTFLAGS=... cargo ...` because it's common and well documented, and people do it all the time. I'm _not_ doing it for cargo-in-cargo, but I don't even know that _you_ are doing cargo-in-cargo?

I don't know what the solution is, but I doubt that silently discarding it is the solution. Maybe we should warn?

---

In other words, there is nothing here that discards your RUSTFLAGS from the top level crate that you are compiling.

---

Fixed. The culprit was RUSTC_WORKSPACE_WRAPPER.